frontend/src/validation-rules.js

```diff
@@ -51,8 +51,19 @@ export const loadAllRules = (i18nCallback) => {
 
   extend('gddUsernameUnique', {
     async validate(value) {
-      const result = await loginAPI.checkUsername(value)
-      return result.result.data.state === 'success'
+      this.$apollo
+        .query({
+          query: checkUsername,
+          variables: {
+            username: value,
+          },
+        })
+        .then((result) => {
+          return result.data.checkUsername.state === 'success'
```

Comment:
state? isn't this obsolete?

Reply:
Yeah we should just return true

Reply:
As this call is not in use yet, we wiill do the refactoring later